packages/vite-plugin/src/index.ts

```diff
@@ -128,10 +134,7 @@ export function vanillaExtractPlugin({
         ssr = ssrParam?.ssr;
       }
 
-      const index = id.indexOf('?');
-      const validId = index === -1 ? id : id.substring(0, index);
-
-      if (ssr) {
+      if (ssr && !process.env.RSC_BUILD) {
```

Comment:
Rather than using an env var, I think this could use an `outputCss` option like the esbuild/webpack plugins, then this line would be `if (ssr || !outputCss) {`. The part I'm missing in this plugin is why it doesn't still go through `processVanillaFile` like the other plugins? cc @mattcompiles.